spec/unicorn_wrangler_spec.rb

```diff
@@ -76,6 +86,31 @@ def with_env(env)
       log = server { get '/slow' }
       expect(log).to include "Garbage collecting: took"
     end
+
+    it "runs custom before hook" do
+      with_env BEFORE_HOOK: '1' do
+        log = server { get }
+        expect(log).to include "GOT BEFORE_HOOK"
+      end
+    end
+
+    it "runs custom hook even when we hijack the hook" do
+      with_env BEFORE_HOOK: '1', FINISH_REQUESTS: '1' do
+        log = server { get }
+        expect(log).to include "GOT BEFORE_HOOK"
+      end
+    end
+
+    it "finishes requests when server receives TERM" do
+      with_env FINISH_REQUESTS: '1' do
+        client = nil
+        # server stays open for 0.1s but client request takes 0.5s
+        server = Thread.new { server { client = Thread.new { get '/vslow' }} }
+        log = server.value
+        client.join
+        expect(log).to include "worker=0 ready"
```

Comment:
Honestly I don't understand this spec. 
Essentially where are we sending TERM to the process?